src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsUtils.kt

```diff
@@ -62,6 +103,30 @@ fun getSurroundingMethod(psiFile: PsiFile, caret: Caret): PsiMethod? {
     return surroundingMethod
 }
 
+/**
+ * Gets the selected line if the constraints on the selected line are satisfied,
+ *   so that EvoSuite can generate tests for it.
+ * Namely, the line is within a method, it is not blank and contains (part of) a statement.
+ *
+ * @param psiFile the current PSI file (where the user makes the click)
+ * @param caret the current (primary) caret that did the click
+ * @return line number if the constraints are satisfied else null
+ */
+fun getSurroundingLine(psiFile: PsiFile, caret: Caret): Int? {
```

Comment:
Maybe this method should be part of the GenerateTestsActionLine instead of here since it's only used there.

Reply:
Similar to `getSurroundingClass` and `getSurroudingMethod`, i have decided to put them here to keep *GenerateTestsAction* classes only for `actionPerformed` and `update` methods (which is why these classes exist). I think it would be more clutter if we add all those methods to the aciton class. Also, this method might be potentially used in other cases (maybe by future developers/contributors? ) and it is not something that is inherent to action class.

Reply:
Sure, fair point